contributingGuides/PROPTYPES_CONVERSION_TABLE.md

```diff
@@ -0,0 +1,131 @@
+# Expensify PropTypes Conversion Table
+
+## Table of Contents
+
+- [Important Considerations](#important-considerations)
+  - [Don't Rely on `isRequired`](#dont-rely-on-isrequired)
+- [PropTypes Conversion Table](#proptypes-conversion-table)
+- [Conversion Example](#conversion-example)
+
+## Important Considerations
+
+### Don't Rely on `isRequired`
+
+Regardless of `isRequired` is present or not on props in `PropTypes`, read through the component implementation to check if props without `isRequired` can actually be optional. The use of `isRequired` is not consistent in the current codebase. Just because `isRequired` is not present, it does not necessarily mean that the prop is optional.
+
+One trick is to mark the prop in question with optional modifier `?`. See if the "possibly `undefined`" error is raised by TypeScript. If any error is raised, the implementation assumes the prop not to be optional.
+
+```ts
+// Before
+const propTypes = {
+  isVisible: PropTypes.bool.isRequired,
+  // `confirmText` prop is not marked as required here, theoretically it is optional.
+  confirmText: PropTypes.string,
+};
+
+// After
+type Props = {
+  isVisible: boolean;
+  // Consider it as required unless you have proof that it is indeed an optional prop.
+  confirmText: string; // vs. confirmText?: string;
+};
+```
+
+## PropTypes Conversion Table
+
+| PropTypes                                                            | TypeScript                                    | Instructions                                                                                                                                                                                                                                  |
+| -------------------------------------------------------------------- | --------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `PropTypes.any`                                                      | `T`, `Record<string, unknown>` or `any`       | Figure out what would be the correct data type and use it.<br><br>If you know that it's a object but isn't possible to determine the internal structure, use `Record<string, unknown>`.                                                       |
+| `PropTypes.array` or `PropTypes.arrayOf(T)`                          | `T[]` or `Array<T>`                           | Convert to `T[]` or `Array<T>`, where `T` is the data type of the array.<br><br>If `T` isn't a primitive type, create a separate `type` for the object structure of your prop and use it.                                                     |
```

Comment:
> Convert to `T[]` or `Array<T>`, where `T` is the data type of the array.

NAB: Didn't we say elsewhere that we should use `T[]` syntax for primitive types and `Array<T>` for custom types? It would be good to state that here as well to be more clear when to use each syntax.

Reply:
Maybe we can simply remove `Array<T>` from the sentence:
```ts
Convert to `T[]`, where `T` is the data type of the array.<br><br>If `T` isn't a primitive type, create a separate `type` for the object structure of your prop and use it. 
```

If `T` isn't primitive -> create a separate `type`. This way, `T[]` will be used in majority of cases. Either way we have a eslint rule for this, I don't think we need to repeat the rule how to use `T[]` vs `Array<T>`